### ISSUE TITLE
Look at this lol

### ISSUE THREAD

**Issue by guarddog13** [bjhiltbrand/drone_mobile_home_assistant#21]
Looks like I'll be looking to pull my Viper Smart Start out and put in a Drone Mobile. I can't for the life of me find a component to get Viper into Home Assistant. I have had it on ST and Hubitat as other people built it out. Their api is undocumented and not easy with refresh tokens etc. So not sure if anyone is willing to assist or not....but with this....I just don't want to spend the money lol.

**Comment by bjhiltbrand** [comment]
I bet that this would act similarly to the Drone Mobile API calls, but as I don't have an account for Viper, I wouldn't be able to do the packet sniffing to see what the API calls look like. If you were able to get a hold of a recorded session with the API calls you are after, I could see if something like this might be possible for HA. Or if you were able to point me to the ST or Hubitat source code for their integrations, I could probably adapt this HA integration to work for Viper as well.

**Comment by guarddog13** [comment]
https://github.com/heidrickla/Hubitat/tree/main/Apps/SmartStart

https://github.com/john159753/rleonard-SmartThings/tree/master/devicetypes/rleonard55/smart-start-2020.src

**Comment by guarddog13** [comment]
There's also this...

https://github.com/vqndev/viper-smart-start-postman/blob/c4da8e80a1b91715334ddac1006e4f00acc20879/vip.php

**Comment by guarddog13** [comment]
So what do you think? Is it possible with the info I gave you?

**Comment by bjhiltbrand** [comment]
Yeah, I think there's a good chance I could work something out. I don't know when I will be able to get to it, but I will post a link to a new project for Viper here when I have something testable.

**Comment by bjhiltbrand** [comment]
How comfortable with Postman are you? Would you mind walking through the API calls from this post:

https://community.smartthings.com/t/release-vehicle-remote-start-smartstart-integration/128183/155

and getting me the response data from each of those calls? I need to see how the data is laid out so I can map it correctly.

Alternatively, if you have the ability to give me temporary credentials to your account, I could get the data that I need. We could also setup a zoom meeting or something similar and I could walk you through getting the data I need. If you want to do that, I have sent you an email with my details so we can connect offline.

**Comment by guarddog13** [comment]
I have got this https://github.com/vqndev/viper-smart-start-postman up and running using the apache2 plug-in on Home Assitant. I'm currently working on the json templates trying to parse the data in Home assistant and it's a real pain in the rear for someone who mostly follows turtorials and copies and pastes. This is where I'm at with it currently. 

https://community.home-assistant.io/t/value-template-error-please-help/441232/28

**Comment by guarddog13** [comment]
If you still have my email could you email me? I've gotten my network fixed, I'm more comfortable with HA. I've moved my HA onto a new computer running on proxmox. I'm quite interested now in figuring out how to create addons and would love to do a zoom thing as we discussed to create this. I've been trying to follow the docs to do this as I've converted the php to a python script. I can't even get the local addons to show up, so I've hit a brick wall.
